### ISSUE TITLE
Transformer outputs probabilities but training uses `BCEWithLogits` (double sigmoid)

### ISSUE THREAD

**Issue by LiuAnclouds** [YXNTU/A-Generalizable-Anomaly-Detection-Method-in-Dynamic-Graphs#6]
Hi, thanks for the great work! I found a potential bug in the training pipeline:

- In `model/Transformer.py`, `TransformerBinaryClassifier.forward` applies a `Sigmoid` before returning:
```
45:83:GeneralDyG-main/model/Transformer.py
class TransformerBinaryClassifier(nn.Module):
    ...
    def forward(self, raw_input, GNN_output, mask):
        ...
        logits = self.classifier(mean_output)
        logits = self.sigmoid(logits).squeeze()
        return logits
```

- But in `train.py`, the loss is computed with `F.binary_cross_entropy_with_logits`, which expects raw logits, not probabilities:
```
26:34:GeneralDyG-main/train.py
def criterion(logits, labels):
    loss_classify = F.binary_cross_entropy_with_logits(logits, labels, reduction='none')
    return torch.mean(loss_classify)
```

This means the model output is passed through `sigmoid` twice (once in the model, once inside `BCEWithLogits`), which can cause saturated gradients and unstable training.

**Comment by YXNTU** [comment]
Thanks for catching this. you're right.

If you remove it inside the model, training should be more stable and maybe even perform better.

Really appreciate the careful check.
